langlib/lang.query/src/main/ballerina/src/lang.query/types.bal

```diff
@@ -34,6 +34,9 @@ type Type any|error;
 @typeParam
 type ErrorType error?;
 
+anydata[][] orderFieldValsArr = [];
+boolean[] orderDirectionsArr = [];
```

Comment:
These arrays are accessed outside _OrderByFunction when adding values to the arrays and sorting. They are emptied in _OrderByFunction so that it won't be an issue when there are multiple queries.